glass/shells.py

```diff
@@ -524,25 +571,53 @@ def partition_nnls(
     dz = np.gradient(zp)
 
     # create the window function matrix
-    a = [np.interp(zp, za, wa, left=0.0, right=0.0) for za, wa, _ in shells]  # type: ignore[arg-type]
-    a /= np.trapz(a, zp, axis=-1)[..., None]  # type: ignore[attr-defined]
+    a = np.array(
+        [
+            np.interp(
+                zp,
+                za,
+                wa,
+                left=0.0,
+                right=0.0,
+            )
+            for za, wa, _ in shells
+        ],
+    )
+    a /= np.trapz(  # type: ignore[attr-defined]
+        a,
+        zp,
+        axis=-1,
+    )[..., None]
```

Comment:
Is that ruff's doing? I really dislike this, it wasn't exactly a long line before...

Reply:
https://github.com/glass-dev/glass/pull/368#discussion_r1831118703